src/aap_eda/tasks/orchestrator.py

```diff
@@ -191,86 +217,106 @@ def dispatch(
+            if request_type != ActivationRequest.RESTART:
+                # A process in PENDING status don't need to update its status.
+                # A monitor can be scheduled for an activation in PENDING
+                # status if its latest process is in workers-offline status
+                # and it is scheduled for restart.
+                if process_parent.status == ActivationStatus.PENDING:
+                    return
+
+                # If the process is in WORKERS_OFFLINE status, it is already
+                # in a bad state.  We don't need to update its status.
+                if process_parent.status == ActivationStatus.WORKERS_OFFLINE:
+                    return
+
+                msg = (
+                    f"{process_parent_type} {process_parent_id} is in an "
+                    "unknown state. The workers of its associated queue "
+                    f"'{queue_name}' are failing liveness checks. "
+                    "There may be an issue with the worker node; "
+                    "please contact the administrator."
+                )
+                status_manager.set_status(
+                    ActivationStatus.WORKERS_OFFLINE,
+                    msg,
+                )
+                status_manager.set_latest_instance_status(
+                    ActivationStatus.WORKERS_OFFLINE,
+                    msg,
+                )
+                LOGGER.warning(msg)
+                return
 
-    if not queue_name:
-        try:
-            queue_name = get_least_busy_queue_name()
-        except HealthyQueueNotFoundError:
-            # The are no healthy queues on which to run the request. For any
-            # request that isn't a restart set its status to "workers offline."
-            msg = (
-                "There are no healthy queues on which to run the request"
-                f"{request_type_text} for {process_parent_type}"
-                f" {process_parent_id}. There may be an issue with the node"
-                "; please contact the administrator."
-            )
+            # The queue is unhealthy, but this is a restart.
+            # The priority is to adhere to the restart policy and
+            # execute the task.
             LOGGER.warning(
-                msg,
+                f"Restarting {process_parent_type} {process_parent_id} "
+                "on the least busy queue; The workers of its associated queue "
+                f"'{queue_name}' are failing liveness checks. "
+                "There may be an issue with the worker node; please contact "
+                "the administrator.",
             )
-
-            if request_type not in [
-                ActivationRequest.RESTART,
-            ]:
-                status_manager = StatusManager(
-                    get_process_parent(process_parent_type, process_parent_id),
+            try:
+                queue_name = get_least_busy_queue_name()
```

Comment:
Here you're trying another queue to proceed restart. A restart needs to stop the old activation instance first. Can it be successfully stopped by another queue?

Reply:
No. This is an exceptional case required as workaround for the incapacity to restart an activation that is running in a unhealthy node. The root problem here is that we can not be 100% sure if the activation is running (excepting for the heartbeat). So the procedure is: set the activation as "workers-offline" so the administrator can try to recover the node or check if the activation can be really restarted. It is the lesser evil. As we can not restart it automatically without the risk of duplicate the activation (which might have catastrophic consequences) at least we allow them to restart it manually. 

In this case, we assume the old pod is cleaned up (or the node is truly dead) and we just run the "restart" operation ignoring the error in the cleanup step.

Reply:
That's fine. My concern is if the stop failed will it block the execution to continue a start.

Reply:
It's not the case, we already skip the exception and continue with the flow. In fact there are more cases where we want to continue with the restart if the cleanup fails, if I remember correctly.